compiler/test/runnable/rvalue1.d

```diff
@@ -139,6 +139,57 @@ void test7()
     StringTest s = StringTest(null);
 }
 
+/********************************/
+// https://github.com/dlang/dmd/issues/20562
+
+struct S8
+{
+    int a,b;
+    this(S8) { printf("this(S)\n"); b = 4; }
+    this(ref S8) { printf("this(ref S)\n"); assert(0); }
+}
+
+
+S8 returnRval(ref S8 arg)
+{
+static if (0)
```

Comment:
what is with these static if branches?

Reply:
Variations on the theme. I used them to try to figure out the best way to resolve the bug. In the comment you'll see the code generated for them.